hack/validate-rover-groups.sh

```diff
@@ -7,18 +7,25 @@ set -o pipefail
 RELEASE_REPO=${RELEASE_REPO:-.}
 
 echo >&2 "$(date -u +'%Y-%m-%dT%H:%M:%S%z') Executing sync-rover-groups validation"
-out=$(sync-rover-groups --manifest-dir=$RELEASE_REPO/clusters --config-file=$RELEASE_REPO/core-services/sync-rover-groups/_config.yaml --log-level=debug --print-config)
 
-# && true avoids the abrupt script termination because diff return code.
-# Exit status is 0 if inputs are the same, 1 if different, 2 if trouble.
-echo "$out" | diff - $RELEASE_REPO/core-services/sync-rover-groups/_config.yaml -U10 && true
-if [[ $? -ne 0 ]]; then
-  echo "ERROR: Changes in release/core-services/sync-rover-groups/_config.yaml ^^^"
-  echo "ERROR: Running sync-rover-groups --validate results in changes"
-  echo "ERROR: To avoid breaking openshift/release for everyone you should"
-  echo "ERROR: fix sync-rover-groups/_config.yaml, removing extra spaces"
-  echo "ERROR: comments and ordering the keys alphabetically."
+# Run sync-rover-groups and capture output
+if ! out=$(sync-rover-groups --manifest-dir=$RELEASE_REPO/clusters --config-file=$RELEASE_REPO/core-services/sync-rover-groups/_config.yaml --log-level=debug --print-config 2>&1); then
+    echo "ERROR: Config file has syntax errors or invalid fields"
+    echo "$out"
+    exit 1
+fi
+
+# Check if normalized output differs from current file
+if ! echo "$out" | diff - $RELEASE_REPO/core-services/sync-rover-groups/_config.yaml -U10 >/dev/null; then
```

Comment:
I wonder if this will work if `diff` returns values different than `0`, maybe we need a `|| true`?